src/integrationtests/java/com/aws/greengrass/integrationtests/lifecyclemanager/GenericExternalServiceIntegTest.java

```diff
@@ -591,13 +591,16 @@ void GIVEN_service_starts_up_WHEN_service_breaks_THEN_status_details_persisted_f
         CountDownLatch serviceErroredLatch = new CountDownLatch(2);
         CountDownLatch serviceBrokenLatch = new CountDownLatch(1);
         List<Pair<ComponentStatusDetails, Long>> componentStatus = new ArrayList<>();
+
+        String serviceName = "ServiceA";
+        CountDownLatch finalServiceErroredLatch = serviceErroredLatch;
```

Comment:
Nit- why do you have these assignments? Could just use two separate latches or reinitialize the original one for your next set of assertions

Reply:
This is required as lambdas must have "effectively final" variables in their captures.

I could create a new variable too, that also works. Re-assigning the value is what breaks the effectively final.